corehq/apps/reports/filters/api.py

```diff
@@ -154,9 +154,40 @@ def extra_data_sources(self):
         # extra data sources to be included for filtering
         raise NotImplementedError('Not implemented yet')
 
+    def custom_locations_search(self):
+        """
+        When the query is specifically searching for just locations and not any other entity like user, group.
+        For ex: enter "/Bihar/patn" would match child locations under locations named Bihar, having name like
+        patn.
+        """
+        return self.q.startswith('/')
+
+    @staticmethod
+    def _get_location_specific_custom_filters(query):
+        query_sections = query.split("/")
```

Comment:
I like this paradigm a lot - we usually display locations as "parent/child", so I'd be in favor of making this the default behavior of `filter_path_by_user_input`, so typing "mass/bos" would match "USA/Massachusetts/Suffolk/Boston"

Reply:
Explained in PR why i chose an exact name search instead of fuzzy search so this change would not help address the issue we are trying to solve here. Though can make that change but it won't be used anywhere as of now.